src/art/skypilot/api.py

```diff
@@ -0,0 +1,203 @@
+            updated_ports = []
+        updated_ports += ["7999", "8000"]
+        resources = resources.copy(ports=updated_ports)
+
+        # check if cluster already exists
+        cluster_status = await to_thread_typed(
+            lambda: sky.status(cluster_names=[self._cluster_name])
+        )
+        if (
+            len(cluster_status) == 0
+            or cluster_status[0]["status"] != sky.ClusterStatus.UP
+        ):
+            await self._launch_cluster(resources, art_version, env_path)
+        else:
+            print(f"Cluster {self._cluster_name} exists, using it...")
+
+        if await is_task_created(
+            cluster_name=self._cluster_name, task_name="art_server"
+        ):
+            print("Art server task already running, using it...")
+        else:
+            art_server_task = sky.Task(name="art_server", run="uv run art")
+            resources = await to_thread_typed(
+                lambda: sky.status(cluster_names=[self._cluster_name])[0][
+                    "handle"
+                ].launched_resources
+            )
+            art_server_task.set_resources(resources)
+
+            # run art server task
+            await to_thread_typed(
+                lambda: sky.exec(
+                    task=art_server_task,
+                    cluster_name=self._cluster_name,
+                    detach_run=True,
+                )
+            )
+            print("Task launched, waiting for it to start...")
+            await wait_for_task_to_start(
+                cluster_name=self._cluster_name, task_name="art_server"
+            )
+            print("Art server task started")
+
+        art_endpoint = await to_thread_typed(
+            lambda: endpoints(cluster=self._cluster_name, port=7999)[7999]
+        )
+        base_url = f"http://{art_endpoint}"
+        print(f"Using base_url: {base_url}")
+
+        super().__init__(base_url=base_url)
+
+    async def _launch_cluster(
+        self,
+        resources: sky.Resources,
+        art_version: str | None = None,
+        env_path: str | None = None,
+    ) -> None:
+        print("Launching cluster...")
+
+        task = sky.Task(
+            name=self._cluster_name,
+        )
+        task.set_resources(resources)
+
+        # TODO: TEST VERSIONED INSTALLATION ONCE WE'VE PUBLISHED A NEW VERSION OF ART WITH THE 'art' CLI SCRIPT
+
+        # default to installing latest version of art
+        art_installation_command = "uv pip install art"
+        if art_version is not None:
+            art_version_is_semver = False
+            # check if art_version is valid semver
+            if art_version is not None:
+                try:
+                    semver.Version.parse(art_version)
+                    art_version_is_semver = True
+                except Exception:
+                    pass
+
+            if art_version_is_semver:
+                art_installation_command = f"uv pip install art=={art_version}"
```

Comment:
nit: `uv add art=={art_version}` would probably work fine too, and give us an error if for some reason it didn't work with awscli or w/e.

Reply:
It might work, but we won't know until we ship a version of art that supports the cli command.